akka-stream/src/main/scala/akka/stream/impl/fusing/GraphStages.scala

```diff
@@ -323,9 +323,9 @@ import akka.stream.stage._
           if (currentIterator.hasNext) {
             if (isAvailable(out)) {
```

Comment:
I think even the `isAvailable(out)` is unnecessary most time, but better to keep it, what to you think @johanandren